ios/Classes/PspdfkitPlugin.m

```diff
@@ -41,23 +46,113 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+            if ([value isEqualToString:@"automatic"]) {
+                userInterfaceMode = PSPDFUserInterfaceViewModeAutomatic;
+            } else if ([value isEqualToString:@"alwaysVisible"]) {
+                userInterfaceMode = PSPDFUserInterfaceViewModeAlways;
+            } else if ([value isEqualToString:@"alwaysHidden"]) {
+                userInterfaceMode = PSPDFUserInterfaceViewModeNever;
+            } else if ([value isEqualToString:@"automaticNoFirstLastPage"]) {
+                userInterfaceMode = PSPDFUserInterfaceViewModeAutomaticNoFirstLastPage;
+            }
+            builder.userInterfaceViewMode = userInterfaceMode;
+        }
+        if (dictionary[@"inlineSearch"]) {
+            builder.searchMode = dictionary[@"inlineSearch"] ? PSPDFSearchModeInline : PSPDFSearchModeModal;
+        }
+        if (dictionary[@"showThumbnailBar"]) {
+            PSPDFThumbnailBarMode thumbnailBarMode = PSPDFThumbnailBarModeScrubberBar;
+            NSString *value = dictionary[@"showThumbnailBar"];
+            if ([value isEqualToString:@"default"]) {
+                thumbnailBarMode = PSPDFThumbnailBarModeScrubberBar;
+            } else if ([value isEqualToString:@"scrollable"]) {
+                thumbnailBarMode = PSPDFThumbnailBarModeScrollable;
+            } else if ([value isEqualToString:@"none"]) {
+                thumbnailBarMode = PSPDFThumbnailBarModeNone;
+            }
+            builder.thumbnailBarMode = thumbnailBarMode;
+        }
+        if (dictionary[@"showPageLabels"]) {
+            builder.pageLabelEnabled = [dictionary[@"showPageLabels"] boolValue];
+        }
+        if (![dictionary[@"enableAnnotationEditing"] boolValue]) {
+            builder.editableAnnotationTypes = nil;
+        }
+        if (dictionary[@"enableTextSelection"]) {
+            builder.textSelectionEnabled = [dictionary[@"enableTextSelection"] boolValue];
+        }
+    }];
+}
+
+- (PSPDFAppearanceMode)PSPDFAppearanceMode:(NSDictionary *)dictionary {
+    if ((id)dictionary == NSNull.null || !dictionary || dictionary.count == 0) {
+        return PSPDFAppearanceModeDefault;
+    }
+    PSPDFAppearanceMode appearanceMode = PSPDFAppearanceModeDefault;
+    if (dictionary[@"appearanceMode"]) {
+        NSString *value = dictionary[@"appearanceMode"];
+        if ([value isEqualToString:@"default"]) {
+            appearanceMode = PSPDFAppearanceModeDefault;
+        } else if ([value isEqualToString:@"night"]) {
+            appearanceMode = PSPDFAppearanceModeNight;
+        } else if ([value isEqualToString:@"sepia"]) {
+            appearanceMode = PSPDFAppearanceModeSepia;
+        }
+    }
+    return appearanceMode;
+}
+
+- (PSPDFPageIndex)PSPDFPageIndex:(NSDictionary *)dictionary {
+    if ((id)dictionary == NSNull.null || !dictionary || dictionary.count == 0) {
+        return 0;
+    }
+    return (PSPDFPageIndex)[dictionary[@"startPage"] unsignedLongValue];
+}
```

Comment:
I'd love if we could create more of those helper methods to extract the actual value from the dictionary, to keep the `PSPDFConfiguration:isImageDocument:` method as clean as possible.

Reply:
Doesn't need to be for all values, but it would at least make sense for enums, like thumbnail bar mode, and user interface view mode.

Reply:
Good idea! Done in https://github.com/PSPDFKit/pspdfkit-flutter/pull/10/commits/f5f7e8c1f5121bbd4f837687d2ca84583a35a213